cpp/CMakeLists.txt

```diff
@@ -786,24 +539,130 @@ set_property(CACHE RMM_LOGGING_LEVEL PROPERTY STRINGS
         "TRACE" "DEBUG" "INFO" "WARN" "ERROR" "CRITICAL" "OFF")
 message(STATUS "RMM_LOGGING_LEVEL = '${RMM_LOGGING_LEVEL}'.")
 
-target_compile_definitions(cudf PUBLIC SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${RMM_LOGGING_LEVEL})
+###################################################################################################
+# - library targets -------------------------------------------------------------------------------
+
+set(CUDF_NAMESPACE "cudf")
+
+# define a function that populates the namespace with modules from the cudf source tree (cpp/src)
+function(add_library_module NAMESPACE MODULE MODULE_SOURCE_DIRS)
+    foreach(MODULE_SOURCE_DIR IN LISTS MODULE_SOURCE_DIRS)
+        file(GLOB_RECURSE MODULE_SOURCE "${MODULE_SOURCE_DIR}/*")
+        list(APPEND MODULE_SRC ${MODULE_SOURCE})
+    endforeach()
+    message(STATUS "Adding library module with name ${MODULE} under namespace ${NAMESPACE} ...")
+    message(STATUS "Using sources: ${MODULE_SRC}")
+    add_library("${NAMESPACE}_${MODULE}" "${MODULE_SRC}")
+    set_target_properties("${NAMESPACE}_${MODULE}" PROPERTIES BUILD_RPATH "\$ORIGIN")
+    # Disable Jitify log printing. See https://github.com/NVIDIA/jitify/issues/79
+    target_compile_definitions("${NAMESPACE}_${MODULE}" PRIVATE "JITIFY_PRINT_LOG=0")
```

Comment:
cc @harrism to confirm we want to add this

Reply:
FYI I added this definition per commit https://github.com/rapidsai/cudf/commit/bb78f56c821f9f5311f6e6bb7273e14057eafd9d